pkg/ec2/getEc2Info.go

```diff
@@ -58,12 +59,23 @@ func DescribeInstance(tagName string, tagValue []string) []EC2Instance {
 	return ec2Instance
 }
 
-func Ec2Client() (context.Context,*ec2.Client) {
+func Ec2Client(cmd *cobra.Command, args []string) (context.Context, *ec2.Client) {
+
 	ctx := context.TODO()
 	cfg, err := config.LoadDefaultConfig(ctx)
 	if err != nil {
 		log.Fatal("Config Error Occured", err)
 	}
+
+	region := cmd.Root().PersistentFlags().Lookup("region").Value.String()
+
+	if region != "" {
+		opts := func(o *ec2.Options) {
```

Comment:
can you explain this function? and also can you add comments wherever necessary in the code.

Reply:
``ec2.NewfromConfig``  takes options in the form of arrays of functions https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#Options